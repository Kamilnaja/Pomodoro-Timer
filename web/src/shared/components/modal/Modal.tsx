import React, { MouseEventHandler } from "react";
import ReactModal from "react-modal";
import LoginContainer from "../../../login/container/LoginContainer";
import Settings from "../../../settings/Settings";
import StatisticsContainer from "../../../statistics/container/StatisticsContainer";
import { Modal as ModalEnum } from "../../store/enums/modal.enum";
import "./modal.scss";

export interface ModalProps {
  modalType: ModalEnum;
  handleCloseModal: MouseEventHandler<HTMLButtonElement>;
}

export const Modal = (props: ModalProps) => (
  <ReactModal
    isOpen={props.modalType !== ModalEnum.NULL}
    ariaHideApp={false}
    shouldFocusAfterRender={true}
    shouldCloseOnOverlayClick={true}
    onRequestClose={props.handleCloseModal}
  >
    <div className="modal">
      <header className="modal__header">
        <button
          className="modal__button--close"
          onClick={props.handleCloseModal}
        >
          âœ•
        </button>
      </header>
      {props.modalType === ModalEnum.LOGIN && <LoginContainer></LoginContainer>}
      {props.modalType === ModalEnum.STATS && (
        <StatisticsContainer></StatisticsContainer>
      )}
      {props.modalType === ModalEnum.SETTINGS && <Settings></Settings>}
    </div>
  </ReactModal>
);
