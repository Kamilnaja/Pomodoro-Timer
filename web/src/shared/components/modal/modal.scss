.modal {
  & &__header {
    height: 25px;
    margin-bottom: 10px;
  }

  & &__button {
    &--close {
      border: none;
      background: none;
      cursor: pointer;
      float: right;
      font-size: 30px;
      margin-top: -20px;
      margin-right: -20px;
      padding: 15px;
    }
  }
}
