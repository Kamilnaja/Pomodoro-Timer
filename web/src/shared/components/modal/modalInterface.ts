export interface CreatedModal {
  component: any;
  title: string;
}
