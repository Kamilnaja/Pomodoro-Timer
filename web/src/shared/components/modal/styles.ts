import { Styles } from "react-modal";

export const customStyles: Styles = {
  content: {
    position: "absolute",
    width: "50%",
    height: "50vh",
    left: "50%",
    top: "50%",
    overflow: "auto",
    backgroundColor: "#fff",
    transform: "translate(-50%, -50%)",
    padding: "40px",
  },
};
