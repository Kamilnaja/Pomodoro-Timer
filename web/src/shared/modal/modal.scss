@import '../styles/shared.scss';

.modal {
  align-items: center;
  background-color: rgba(0, 0, 0, 0.7) !important;
  bottom: 0;
  display: flex !important;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 999;

  & &__header {
    display: flex;
    justify-content: space-between;
  }

  & &__title {
    @include font-fira-sans($size: 25px, $weight: bold);
    padding: 10px 0;
  }

  & &__button {
    &--close {
      background: none;
      border: none;
      cursor: pointer;
      font-size: 30px;
    }
  }

  & &__content {
    background-color: #fff;
    max-height: 100vh;
    overflow: auto;
    padding: 20px 40px 40px;
    width: 60%;

    &--statistics {
      height: 100%;
    }
  }
}

@media (max-width: $sm) {
  .modal {
    & &__content {
      padding: 20px;
      width: 100%;
    }
  }
}
