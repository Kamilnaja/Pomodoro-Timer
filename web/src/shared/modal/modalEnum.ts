export enum ModalType {
  LOGIN,
  REGISTER,
  LOGOUT,
  STATS,
  NULL,
}
