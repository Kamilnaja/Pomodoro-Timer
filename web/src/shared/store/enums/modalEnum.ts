export enum Modal {
  LOGIN,
  REGISTER,
  SETTINGS,
  LOGOUT,
  NULL,
}
