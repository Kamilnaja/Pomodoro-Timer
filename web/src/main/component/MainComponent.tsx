import HeaderContainer from '../../header/container/HeaderContainer';
import { MainComponentProps } from './MainComponentProps';
import './mainComponent.scss';
import { Modal } from '../../shared/modal/Modal';
import CookiesInfoContainer from '../../shared/cookieInfo/container/CookiesInfoContainer';

export const MainComponent = (props: MainComponentProps) => {
  return (
    <div className="app">
      <HeaderContainer handleOpenModal={props.handleOpenModal} isLoggedIn={props.isLoggedIn} />
      <Modal modalType={props.openedModal} closeModal={props.handleCloseModal} />
      <CookiesInfoContainer />
    </div>
  );
};
