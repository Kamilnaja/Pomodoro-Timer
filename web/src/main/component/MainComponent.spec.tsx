import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ModalType } from '../../shared/modal/modalEnum';
import { MainComponent } from './MainComponent';

describe('MainComponent', () => {
  let wrapper: ShallowWrapper;
  const handleCloseModal = jest.fn();
  const handleOpenModal = jest.fn();
  const openedModal = ModalType.LOGIN;

  beforeEach(() => {
    wrapper = shallow(
      <MainComponent
        handleCloseModal={handleCloseModal}
        handleOpenModal={handleOpenModal}
        openedModal={openedModal}
        isLoggedIn={false}
      />,
    );
  });

  it('should works', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
