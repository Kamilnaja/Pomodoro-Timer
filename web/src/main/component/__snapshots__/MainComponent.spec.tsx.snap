// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent should works 1`] = `
<div
  className="app"
>
  <HeaderContainer
    handleOpenModal={[MockFunction]}
    isLoggedIn={false}
  />
  <Modal
    closeModal={[MockFunction]}
    modalType={0}
  />
  <Connect(CookiesInfoContainer) />
</div>
`;
